### PR TITLE
feat: implement synchronization cross-check deadline with schedule and dashboard

### PR DESCRIPTION
**Pull request by NikitaKorevo**
**🟢 Add `deploy` label if you want to deploy this Pull Request to staging environment**

#### 🧑‍⚖️ Pull Request Naming Convention

- Title should follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- Do not put issue id in title
- Do not put WIP in title. Use [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) functionality
- Consider to add `area:*` label(s)

* [x] I followed naming convention rules

---

#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related issue link

[#1521](https://github.com/rolling-scopes/rsschool-app/issues/1521)
[#1378](https://github.com/rolling-scopes/rsschool-app/issues/1378)
[#1045](https://github.com/rolling-scopes/rsschool-app/issues/1045)

#### 💡 Background and solution

Если у таски есть свойство `сrossCheckEndDate`, то добавляется строка с датой окончания кросс-чека в Schedule и Dashboard.

Schedule:
schedule

Dashboard:
dashboard

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Database migration is added or not needed
- [x] Documentation is updated/provided or not needed
- [x] Changes are tested locally

### CODE REVIEW

**Comment by apalchys** [comment]
@NikitaKorevo  thanks for your changes. 

There are some changes pending that will affect how Schedule works. We decided to rework and create a new API for schedule as current becomes messy and it's hard to support. There is open PR in draft state that you can check
https://github.com/rolling-scopes/rsschool-app/pull/1647

it adds new API in nestjs for getting schedule items. Could you please postpone your changes and update PR when I merge my PR ?

**Comment by NikitaKorevo** [comment]
@apalchys хорошо, перевел пулл реквест в драфт.
Может мой пулл реквест отменить? Как мне кажется, лучше будет сразу в твоем пулл реквесте синхронизацию кросс-чек дедлайна добавить.

File: client/src/pages/course/student/dashboard.tsx [hunk 1]
```diff
@@ -212,7 +213,20 @@ const tasksToEvents = (tasks: CourseTaskDto[]) => {
   return tasks.reduce((acc: Array<CourseEvent>, task: CourseTaskDto) => {
     if (task.type !== TaskTypes.test) {
       acc.push(createCourseEventFromTask(task, task.type));
+
+      if (task.crossCheckEndDate) {
+        const crossCheckDeadlineEvent: CourseTaskDto = {
+          ...task,
+          id: task.id + 0.5,
```
Comment (AlreadyBored) [inline client/src/pages/course/student/dashboard.tsx:220]: Could you please explain what magic happens here?

File: nestjs/src/cross-check/tasks-filtering.ts [hunk 2]
```diff
@@ -8,6 +8,6 @@ export const isTaskNeededToStart = ({ crossCheckStatus, studentEndDate }: Course
 
 export const isTaskNeededToFinish = ({ crossCheckStatus, crossCheckEndDate }: CourseTask) => {
   const currTimestampUTC = Date.now();
-  const crossCheckEndDateTimestampUTC = Date.parse(crossCheckEndDate);
+  const crossCheckEndDateTimestampUTC = Date.parse(crossCheckEndDate as string);
```
Comment (AlreadyBored) [inline nestjs/src/cross-check/tasks-filtering.ts:11]: Please consider to avoid typecast here

File: client/src/pages/course/admin/tasks.tsx [hunk 3]
```diff
@@ -242,7 +242,7 @@ function Page(props: CoursePageProps) {
           <DatePicker.RangePicker format="YYYY-MM-DD HH:mm" showTime={{ format: 'HH:mm' }} />
         </Form.Item>
         {modalData?.checker === 'crossCheck' ? (
-          <Form.Item name="crossCheckEndDate" label="Cross-Check End Date">
+          <Form.Item name="crossCheckEndDate" label="Cross-Check End Date" rules={[{ required: true }]}>
```
Comment (AlreadyBored) [inline client/src/pages/course/admin/tasks.tsx:245]: Why should it be mandatory?

File: nestjs/src/courses/course-tasks/dto/course-task.dto.ts [hunk 4]
```diff
@@ -26,6 +26,7 @@ export class CourseTaskDto {
     this.name = courseTask.task.name;
     this.studentStartDate = (courseTask.studentStartDate as Date)?.toISOString();
     this.studentEndDate = (courseTask.studentEndDate as Date)?.toISOString();
+    this.crossCheckEndDate = (courseTask.crossCheckEndDate as Date)?.toISOString();
```
Comment (AlreadyBored) [inline nestjs/src/courses/course-tasks/dto/course-task.dto.ts:29]: Please, consider to avoid typecast here

**Comment by apalchys** [comment]
> Может мой пулл реквест отменить? Как мне кажется, лучше будет сразу в твоем пулл реквесте синхронизацию кросс-чек дедлайна добавить.

Да, наверное лучше отменить. Issue можно заасайнить на меня.

**Comment by NikitaKorevo** [comment]
@AlreadyBored отменил пулл-реквест, но спасибо за ревью

**Pull request closed without merging**